native-addons-are-close-to-being-on-par-with-js-modules.md

```diff
@@ -0,0 +1,113 @@
+
+```
+Error: The module '/home/user/node_hello_world/build/Release/hello.node'
+was compiled against a different Node.js version using
+NODE_MODULE_VERSION 57. This version of Node.js requires
+NODE_MODULE_VERSION 64. Please try re-compiling or re-installing
+the module (for instance, using `npm rebuild` or `npm install`).
+    at Object.Module._extensions..node (internal/modules/cjs/loader.js:718:18)
+    at Module.load (internal/modules/cjs/loader.js:599:32)
+    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
+    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
+    at Module.require (internal/modules/cjs/loader.js:637:17)
+    at require (internal/modules/cjs/helpers.js:22:18)
+    at bindings (/home/user/node_hello_world/node_modules/bindings/bindings.js:76:44)
+    at Object.<anonymous> (/home/user/node_hello_world/hello.js:1:94)
+    at Module._compile (internal/modules/cjs/loader.js:689:30)
+    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
+```
+
+Such errors can also be avoided by the use of prebuild utilities, however, it places even more burden on addon maintainers, namely, to rebuild the binaries they provide at every release of Node.js. It also prevents application maintainers from simply dropping a new version of Node.js into their production environment. Indeed they need to reinstall their application even though no part of its functionality has changed.
+
+[N-API][] was introduced to address the problem of the binary compatibility of native addons with versions of Node.js against which they were build *as well as all subsequent versions*. It abstracts away essential JavaScript engine and Node.js APIs, presenting them to native addon maintainers as a collection of ABI-stable APIs. The stability of N-API is assured by the fact that all APIs are C APIs rather than C++ APIs and by a commitment on the part of the Node.js core developer community to avoid modifying the interface in an incompatible way.
+
+N-API eliminates one of the three variables that causes the proliferation of binaries required to provide a single Node.js native addon: the integer value that represents the Node.js version against which the binary is designed to work (`NODE_MODULE_VERSION`).
```

Comment:
I may be confused here. Do you mean `NAPI_VERSION` instead of `NODE_MODULE_VERSION`?

Reply:
I do mean `NODE_MODULE_VERSION`. N-API modules are unaffected by changes to that number.

Reply:
👍🏼